src/OrleansProviders/SQLServer/CreateTables.sql

```diff
@@ -1,113 +1,1440 @@
+   long as the interface contract is maintained.
+     
+4) The implementation across vendor specific scripts *should* preserve the constraint names. This simplifies troubleshooting
+   by virtue of uniform naming across concrete implementations.
+
+5) ETag or VersionETag for Orleans is an opaque BINARY(n) or VARBINARY(n) column that Orleans transforms to a string and back to BINARY(n) or
+   VARBINARY(n) when querying. The type of its actual implementation is not important.
+
+6) For the sake of being explicit and removing ambiquity, Orleans expects some queries to return either TRUE or FALSE as an
+   indication of success. Orleans reads this value as ADO.NET Boolean value.
+   That is, affected rows or such does not matter. If an error is raised or an exception is thrown
+   the query *must* ensure the entire transaction is rolled back and may either return FALSE or propagate the exception.
+   Orleans handles exception as a failure and will (likely) retry.
+
+   Additional note: along with the boolean success value other information could be provided too, such as an ETag
+   of the operated entity and/or error codes equivalent to HTTP error codes.
+
+   The operations *must* succeed atomically as mandated by Orleans membership protocol. For more
+   information, see at
+		http://dotnet.github.io/orleans/Runtime-Implementation-Details/Runtime-Tables.html
+		https://github.com/dotnet/orleans/blob/master/src/Orleans/SystemTargetInterfaces/IMembershipTable.cs
+*/
+
+
+-- Information of this view can be used to tune queries in database and deployment specific ways if needed.
+CREATE VIEW [OrleansDatabaseInfo] AS
+-- Version information derived from https://support.microsoft.com/en-us/kb/321185.
+SELECT
+    N'ProductName' AS [Id],
+    CASE 
+        WHEN LEFT(CAST(SERVERPROPERTY('ProductVersion') AS NVARCHAR), CHARINDEX('.', CAST(SERVERPROPERTY('ProductVersion') AS NVARCHAR)) - 1) = 8 THEN 'SQL Server 2000'
+        WHEN LEFT(CAST(SERVERPROPERTY('ProductVersion') AS NVARCHAR), CHARINDEX('.', CAST(SERVERPROPERTY('ProductVersion') AS NVARCHAR)) - 1) = 9 THEN 'SQL Server 2005'
+        WHEN LEFT(CAST(SERVERPROPERTY('ProductVersion') AS NVARCHAR), CHARINDEX('.', CAST(SERVERPROPERTY('ProductVersion') AS NVARCHAR)) - 1) = 10 
+            AND RIGHT(LEFT(CAST(SERVERPROPERTY('ProductVersion') AS NVARCHAR), 5), 2) = 00 THEN 'SQL Server 2008'
+        WHEN LEFT(CAST(SERVERPROPERTY('ProductVersion') AS NVARCHAR), CHARINDEX('.', CAST(SERVERPROPERTY('ProductVersion') AS NVARCHAR)) - 1) = 10 
+            AND RIGHT(LEFT(CAST(SERVERPROPERTY('ProductVersion') AS NVARCHAR), 5), 2) = 50 THEN 'SQL Server 2008 R2' 
+        WHEN LEFT(CAST(SERVERPROPERTY('ProductVersion') AS NVARCHAR), CHARINDEX('.', CAST(SERVERPROPERTY('ProductVersion') AS NVARCHAR)) - 1) = 11 THEN 'SQL Server 2012'
+        WHEN LEFT(CAST(SERVERPROPERTY('ProductVersion') AS NVARCHAR), CHARINDEX('.', CAST(SERVERPROPERTY('ProductVersion') AS NVARCHAR)) - 1) = 12 THEN 'SQL Server 2014' 				
```

Comment:
SQL Server 2016 is missing. :) I haven't checked if SQL Server Azure has something specific in it.